test/e2e/provision.go

```diff
@@ -0,0 +1,90 @@
+package e2e
+
+import (
+	"context"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"os"
+	"path"
+	"sigs.k8s.io/e2e-framework/klient/decoder"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"time"
+)
+
+// CloudProvision defines operations to provision the environment on cloud providers.
+type CloudProvision interface {
+	CreateCluster(ctx context.Context, cfg *envconf.Config) error
+	CreateVPC(ctx context.Context, cfg *envconf.Config) error
+}
+
+type PeerPods struct {
+	cloudProvider string
+	namespace     string
+}
+
+func NewPeerPods(provider string) (p *PeerPods) {
+	return &PeerPods{cloudProvider: provider,
+		namespace: "confidential-containers-system"}
+}
+
+func (p *PeerPods) Delete(ctx context.Context, cfg *envconf.Config) error {
+	// TODO: implement me.
+	return nil
+}
+
+// Deploy installs Peer Pods on the cluster.
+func (p *PeerPods) Deploy(ctx context.Context, cfg *envconf.Config) error {
+	client, err := cfg.NewClient()
+	if err != nil {
+		return err
+	}
+
+	err = decoder.DecodeEachFile(ctx, os.DirFS("../../install/yamls"), "deploy.yaml", decoder.CreateIgnoreAlreadyExists(client.Resources()))
+	if err != nil {
+		return err
+	}
+
+	if err := AllPodsRunning(ctx, cfg, p.namespace); err != nil {
+		return err
+	}
+
+	overlayDir := path.Join("../../install/overlays", p.cloudProvider)
+	kustomizeHelper := &KustomizeHelper{configDir: overlayDir}
+	if err := kustomizeHelper.Apply(ctx, cfg); err != nil {
+		return err
+	}
+
+	if err := AllPodsRunning(ctx, cfg, p.namespace); err != nil {
```

Comment:
Looks need also make sure RuntimeClasses ready here.

Reply:
hmm... I will fix it.